Spatial.gd

```diff
@@ -319,6 +328,27 @@ func decode_context_packet(spb: StreamPeerBuffer):
 	# vs radians. 70deg = 1.22173rad and 25deg = 0.4363323rad. We should redo
 	# this to just be a radian to degree conversion.
 
+	# Calculations to dynamically place the main icon/button
+	# The left- and right-margin can be calculated from the ui-scale combined with the preset icon width and the desired position.
+	# set_minimal_mouse_block() should be called only once, if it is called while any burrito windows is open it will become unclickable.
+	ui_size = identity["uisz"]
+	# If the value is outside of the expected range use the "normal" size.
+	if (ui_size < 0) or (ui_size > 3):
+		ui_size = 1
+
+	var button_position_max = floor(OS.window_size.x / icon_size_preset[ui_size]) - 1
```

Comment:
I'm not sure this is the correct solution. Guild Wars 2 seems to shrink the UI in small windows. I have some examples below. 

![image](https://github.com/AsherGlick/Burrito/assets/89052698/474e8f52-ebf5-486d-89b6-d15c7079e7ca)
![image](https://github.com/AsherGlick/Burrito/assets/89052698/3600b6ff-80e9-436a-bc42-821f78a7edda)

Reply:
You are right I did not think about a movable window, my point of view was a borderless fullscreen window.
If the position for our button ends up outside the window (for whatever reason) it would not be visible, with that limit it would be appear in the right corner of the screen regardless.

I will try to find a way to keep the button inside the window even if it is moved around.

Reply:
IIRC there was some discussion on the blishud discord at some point about how this sizing algorithm works. I cant remember where but I think it is documented... somewhere.

Reply:
Ok, I thought about this and come to the conclusion that running GW2 in a partial window needs a lot of work outside the scope of this PR.
 - The position of burrito does not follow the GW2 window
 - We need to work in an offset to compensate the titlebar
 - Dynamic scaling of elements

![Screenshot_20231120_221219](https://github.com/AsherGlick/Burrito/assets/30129213/f4734eee-af97-45e1-898b-201d7aab2f29)

So, we can either keep this check, because it does what I want it to do as long as GW2 is running fullscreen or we remove it because it is not useful.

Reply:
If it were my pr I would just remove this check and assume the window is big enough, then revisit it when we have a sticky position